src/metalookup/core/metadata_manager.py

```diff
@@ -90,43 +88,18 @@ async def extract(self, message: Input, extra: bool) -> Output:
         """
         self.logger.debug("Calling extractors from manager")
 
-        # this will eventually load the content dynamically if not provided in the message
-        # hence it may fail with various different exceptions (ConnectionError, ...)
-        # those exceptions will be handled below.
-        try:
-            with runtime() as t:
-                site = await WebsiteData.from_input(
-                    input=message,
-                    logger=self.logger,
-                    tld_extractor=self.tld_extractor,
-                )
-            self.logger.info(f"Built WebsiteData object in {t():5.2f}s.")
-            # fixme: For now we simply assume that the last entry is the relevant one.
-            #        This may be fixed together with https://github.com/openeduhub/metalookup/issues/85
-            response_code = site.har.log.entries[-1].response.status
-            if response_code != 200:
-                raise HTTPException(
-                    status_code=502, detail=f"Resource could not be loaded by splash (reported: {response_code})"
-                )
-
-        # Technically, for the user the communication with the splash container is
-        # an implementation detail of the server. Returning a 502 (Bad Gateway)
-        # should indicate to the user, that the resource (in this case the url
-        # that was transmitted to the extract endpoint) is no longer available,
-        # whereas internal communication problems should not give this indication
-        # to the user.
-        except ClientError as e:
-            raise HTTPException(status_code=500, detail=f"Could not get HAR from splash: {e}")
-        except asyncio.exceptions.TimeoutError:
-            raise HTTPException(status_code=500, detail="Splash container request timeout.")
-        except ValidationError as e:
-            raise HTTPException(status_code=500, detail=f"Received unexpected HAR from splash: {e}")
+        if message.splash_response is None:
+            content = Content(url=message.url)
```

Comment:
Yes the task will return the response from Splash once complete. 

It is not a singleton, because for every incoming extraction request there will be exactly one `Content` object. This single `content` object however is shared for all the extraction tasks of that request. Hence the `extactors` also share the respective task / splash response.

Reply:
Is the sharing magic in the _task function, i.e., because you await the task connected to the url?

Reply:
I would argue, the sharing happens in the MetadataManager. 
 - There, the content instance is created and passed to all the extraction tasks. 
 - From within those tasks the har or html may or may not be accessed. 
 - However once one tasks accessess har or html, it will be available for all other extractors as well, because there is only one task (not to be confused with the extraction tasks) to fetch from splash.